### DISCUSSION TITLE
🐛 Bug: SQLIte : Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?

### DISCUSSION THREAD

**Discussion by from-nibly** [nocodb/nocodb#11863] [Self hosting]
### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

Not entirely sure, but I'm running docker with sqlite. nocodb version is 202.10.

### Desired Behavior

Not to timeout?

### Project Details

Node: **v18.18.2**
Arch: **x64**
Platform: **linux**
Docker: **false**
RootDB: **sqlite3**
PackageVersion: **0.202.10**

### Attachments

After a while it loops printing this

```
nocodb-0 nocodb Error: Formula error: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
nocodb-0 nocodb     at BaseModelSqlv2.getSelectQueryBuilderForFormula (/usr/src/app/docker/main.js:2:505586)
nocodb-0 nocodb     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
nocodb-0 nocodb     at async BaseModelSqlv2.selectObject (/usr/src/app/docker/main.js:2:510768)
nocodb-0 nocodb     at async BaseModelSqlv2.readByPk (/usr/src/app/docker/main.js:2:482367)
nocodb-0 nocodb     at async BaseModelSqlv2.bulkUpdate (/usr/src/app/docker/main.js:2:523511)
nocodb-0 nocodb     at async DataTableService.dataUpdate (/usr/src/app/docker/main.js:2:1681995)
nocodb-0 nocodb     at async DataTableController.dataUpdate (/usr/src/app/docker/main.js:2:374754)
```

maybe it's one of the formulas?

**Comment by from-nibly** [comment]
I deleted one of the columns with a formula and recreated it, it ended up fixing the issue.

**Comment by CheetahAzure** [comment]
I encountered this problem too when migrating the sqlite database to version 0.204.1

**Comment by dstala** [comment]
Fixed in v0.205.0
Please switch over to latest NocoDB. Let us know if issue persists

**Comment by biancamoldovanm** [comment]
Hello @dstala, the issue is happening on the latest version too. (v0.262.1)

**Comment by cfab** [comment]
Same here... my nocodb instance regularly stop functionnning and I get this in the logs. It works on a cloudron server. Can it be related to lack of memory (I only have  2gb) ?

**Comment by cfab** [comment]
@dstala  can you reopen the issue ?

**Comment by o1lab** [comment]
We strongly advise using postgres with NocoDB as we no longer can support all underlying databases as NocoDb's default database.
